backend/src/resolvers/comments.js

```diff
@@ -36,20 +40,84 @@ export default {
       if (!post) {
         throw new UserInputError(NO_POST_ERR_MESSAGE)
       }
-      const comment = await neo4jgraphql(object, params, context, resolveInfo, false)
 
-      await session.run(`
+      const comment = await neo4jgraphql(
+        object,
+        params,
+        context,
+        resolveInfo,
+        false
+      )
+
+      await session.run(
+        `
         MATCH (post:Post {id: $postId}), (comment:Comment {id: $commentId}), (author:User {id: $userId})
         MERGE (post)<-[:COMMENTS]-(comment)<-[:WROTE]-(author)
-        RETURN post`, {
-        userId: context.user.id,
-        postId,
-        commentId: comment.id
-      }
+        RETURN post`,
+        {
+          userId: context.user.id,
+          postId,
+          commentId: comment.id
+        }
       )
       session.close()
 
       return comment
+    },
+    UpdateComment: async (object, params, context, resolveInfo) => {
+      // Strip element tags and remove leading/trailing white spaces from content
+      const content = params.content.replace(/<(?:.|\n)*?>/gm, '').trim()
+      const { id } = params
+      // Check length of content
+      if (!params.content || content.length < COMMENT_MIN_LENGTH) {
+        throw new UserInputError(
+          `Comment must be at least ${COMMENT_MIN_LENGTH} character long!`
+        )
+      }
+
+      // Check if comment exists
+      const session = context.driver.session()
+      const commentQueryRes = await session.run(
+        `
+        MATCH (comment: Comment { id:$id}) 
+        RETURN comment`,
+        {
+          id
+        }
+      )
+
+      // Destructure content from session results array
+      const [comment] = commentQueryRes.records.map(record => {
+        return record.get('comment')
+      })
+
+      // Send error message if cannot find a matching comment.
+      if (!comment) {
+        throw new UserInputError(NO_COMMENT_ERR_MESSAGE)
+      }
+
+      // Update comment.
+      const commentRev = await neo4jgraphql(
+        object,
+        params,
+        context,
+        resolveInfo,
+        false
+      )
+
+      await session.run(
+        `
+        MATCH (comment: Comment { id:$id}) 
+        SET comment.content = $content
+        `,
+        {
+          id,
+          content
+        }
+      )
+      session.close()
```

Comment:
I'd be interested to know more about what you are talking about @roschaefer, I'm not sure I understood what you are trying to say... :thinking: 

It would be great to see some backend tests though @ALau2088, but so far it's looking good... great job!!

Reply:
@ALau2088 first checks if the comment exists, then sending an error message in case, then updating the comment, then `MATCH`ing the comment again to update the `content` - I find it cumbersome and I wanted to suggest to check if there is an easier way?

Why not simply:

```
      params.content = params.content.replace(/<(?:.|\n)*?>/gm, '').trim()
```
at line 69?

I guess you can remove everything after
```
await neo4jgraphql(
        object,
        params,
        context,
        resolveInfo,
        false
      )
```

Reply:
great! that makes sense @roschaefer